src/Chapter2.hs

```diff
@@ -841,7 +871,14 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate a arr
+  | a < 0 = []
+  | length arr == 0 = []
```

Comment:
`length` is a slow function for lists, and it's better to use `null` to check lists on emptiness
```suggestion
  | null arr = []
```